service/storage_service.cc

```diff
@@ -830,6 +830,68 @@ void storage_service::scan_cdc_generations() {
     }
 }
 
+future<> storage_service::check_and_repair_cdc_streams() {
+    return async([this] { 
+        auto latest = _cdc_streams_ts;
+        const auto& endpoint_states = _gossiper.get_endpoint_states();
+        for (const auto& [addr, state] : endpoint_states) {
+            if (!_gossiper.is_normal(addr))  {
+                throw std::runtime_error(format("All nodes must be in NORMAL state while performing check_and_repair_cdc_streams"
+                        " ({} is in state {})", addr, _gossiper.get_gossip_status(state)));
+            }
+
+            const auto ts = cdc::get_streams_timestamp_for(addr, _gossiper);
+            if (!latest || (ts && *ts > *latest)) {
+                latest = ts;
+            }
+        }
+
+        bool should_regenerate = false;
+        const auto gen = _sys_dist_ks.local().read_cdc_topology_description(
+                *latest, { _token_metadata.count_normal_token_owners() }).get0();
```

Comment:
One thing I would add there.
`read_cdc_topology_description` might  throw and error, for example when:
- replicas are unreachable due to a network partition: timeout, unavailable exception
- the table is corrupted: stuff like "read expected size mismatch", see https://github.com/scylladb/scylla/issues/6537

There is an expected class of errors (timeout etc.) and there is an unexpected class of errors that occur due to bugs (table corruption etc.).

In the expected class we should return the error to the user and finish the procedure.
So, for example, if replicas are unreachable due to a network partition, so `read_cdc_topology_description` throws a timeout, we should inform the user and return.
Although that's debatable. Maybe it would be safer to continue and create a new generation nevertheless.

In the unexpected class we should definitely create a new generation. For example, if the latest generation entry is corrupted in the table, CDC won't work unless we create a new gen.

So, the `read_cdc_topology_description` call should be wrapped in a `try...catch`, we should catch unexpected errors such as `read_simple_exactly - size mismatch`, report them using `cdc_log.error(...)`, set `should_regenerate = true` and continue.